### PR TITLE
Use .NET APIs to Simplify Hashing Code

### PR DESCRIPTION
**Pull request by slang25**
Now we are using .NET 5 (see #27), we can use some of the nice new convienience APIs 👌

### CODE REVIEW

File: Functions.Tests/Functions.Tests.csproj [hunk 1]
```diff
@@ -5,6 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.10.0" />
```
Comment (slang25) [inline Functions.Tests/Functions.Tests.csproj:8]: I see @stebet spotted the same thing in #24, I'll revert this commit here in favour of this being addressed in that PR

Comment (stebet) [reply Functions.Tests/Functions.Tests.csproj]: > Now we are using .NET 5 (see #27), we can use some of the nice new convienience APIs 👌

We're not on .NET 5 yet :) This is still targeting .NET 3.1

Comment (stebet) [reply Functions.Tests/Functions.Tests.csproj]: B.t.w, did a benchmark against some change I have in the pipes, and although the code is cleaner, it does regress in performance:

```ini
BenchmarkDotNet=v0.13.0, OS=Windows 10.0.21390
Intel Core i7-10700 CPU 2.90GHz, 1 CPU, 16 logical and 8 physical cores
.NET SDK=5.0.300
  [Host]     : .NET 5.0.6 (5.0.621.22011), X64 RyuJIT
  DefaultJob : .NET 5.0.6 (5.0.621.22011), X64 RyuJIT
```

|  Method |                Input |     Mean |   Error |  StdDev | Ratio | RatioSD |  Gen 0 | Gen 1 | Gen 2 | Allocated |
|-------- |--------------------- |---------:|--------:|--------:|------:|--------:|-------:|------:|------:|----------:|
|  Before |            Passw0rd! | 326.0 ns | 2.94 ns | 2.75 ns |  1.00 |    0.00 | 0.0410 |     - |     - |     344 B |
|   After |            Passw0rd! | 288.3 ns | 3.55 ns | 2.96 ns |  0.88 |    0.01 | 0.0124 |     - |     - |     104 B |
| Slang25 |            Passw0rd! | 395.6 ns | 5.00 ns | 4.44 ns |  1.21 |    0.02 | 0.0229 |     - |     - |     192 B |
|         |                      |          |         |         |       |         |        |       |       |           |
|  Before | ThisI(...)Bytes [89] | 454.6 ns | 5.70 ns | 5.33 ns |  1.00 |    0.00 | 0.0505 |     - |     - |     424 B |
|   After | ThisI(...)Bytes [89] | 416.6 ns | 3.88 ns | 3.63 ns |  0.92 |    0.01 | 0.0124 |     - |     - |     104 B |
| Slang25 | ThisI(...)Bytes [89] | 511.9 ns | 5.01 ns | 3.91 ns |  1.13 |    0.02 | 0.0324 |     - |     - |     272 B |
|         |                      |          |         |         |       |         |        |       |       |           |
|  Before |              hunter2 | 328.0 ns | 3.40 ns | 3.18 ns |  1.00 |    0.00 | 0.0401 |     - |     - |     336 B |
|   After |              hunter2 | 286.6 ns | 3.03 ns | 2.69 ns |  0.87 |    0.01 | 0.0124 |     - |     - |     104 B |
| Slang25 |              hunter2 | 391.0 ns | 3.79 ns | 3.16 ns |  1.19 |    0.02 | 0.0219 |     - |     - |     184 B |

Comment (stebet) [reply Functions.Tests/Functions.Tests.csproj]: I do really like the shortened code though, let's see if we can somehow take the best of both worlds here :)

Comment (mdawsonuk) [reply Functions.Tests/Functions.Tests.csproj]: I must say that keeping the code as fast as possible is absolutely key for the password ingest pipeline which is in the works

Comment (slang25) [reply Functions.Tests/Functions.Tests.csproj]: #27 has been merged into `main`, if we were on 3.1 this wouldn't compile.

Thanks for running the benchmark. That's a little disappointing, I thought that the BCL implementation would have more potential to squeeze performance gains, but clearly some of the community found solutions still win on both latency and heap allocations. I'll do a bit of investiging. I wonder if it's the hashing or the hex string formatting, or both.

Comment (stebet) [reply Functions.Tests/Functions.Tests.csproj]: Been trying a few iterations, the convenience methods are all considerably slower :(

Comment (stebet) [reply Functions.Tests/Functions.Tests.csproj]: Ahh didn't see #27 had been merged. Thanks for the heads up :)

Comment (troyhunt) [reply Functions.Tests/Functions.Tests.csproj]: At present, 1 in 228 queries actually hits Azure and we're talking about a *very* small change in speed. Plus, I just realised this code isn't actually being hit anyway so yeah, let's just do it 😎

**Pull request merged**